allensdk/brain_observatory/behavior/data_objects/eye_tracking/rig_geometry.py

```diff
@@ -0,0 +1,220 @@
+
+    @property
+    def equipment(self):
+        return self._equipment
+
+    def to_nwb(self, nwbfile: NWBFile) -> NWBFile:
+        eye_tracking_rig_mod = pynwb.ProcessingModule(
+            name='eye_tracking_rig_metadata',
+            description='Eye tracking rig metadata module')
+
+        nwb_extension = load_pynwb_extension(
+            OphysEyeTrackingRigMetadataSchema, 'ndx-aibs-behavior-ophys')
+
+        rig_metadata = nwb_extension(
+            name="eye_tracking_rig_metadata",
+            equipment=self._equipment,
+            monitor_position=self._monitor_position_mm,
+            monitor_position__unit_of_measurement="mm",
+            camera_position=self._camera_position_mm,
+            camera_position__unit_of_measurement="mm",
+            led_position=self._led_position,
+            led_position__unit_of_measurement="mm",
+            monitor_rotation=self._monitor_rotation_deg,
+            monitor_rotation__unit_of_measurement="deg",
+            camera_rotation=self._camera_rotation_deg,
+            camera_rotation__unit_of_measurement="deg"
+        )
+
+        eye_tracking_rig_mod.add_data_interface(rig_metadata)
+        nwbfile.add_processing_module(eye_tracking_rig_mod)
+        return nwbfile
+
+    @classmethod
+    def from_nwb(cls, nwbfile: NWBFile) -> "RigGeometry":
+        try:
+            et_mod = \
+                nwbfile.get_processing_module("eye_tracking_rig_metadata")
+        except KeyError as e:
+            warnings.warn("This ophys session "
+                          f"'{int(nwbfile.identifier)}' has no eye "
+                          f"tracking rig metadata. (NWB error: {e})")
+            raise
+
+        meta = et_mod.get_data_interface("eye_tracking_rig_metadata")
+
+        monitor_position = meta.monitor_position[:]
+        monitor_position = (monitor_position.tolist()
+                            if isinstance(monitor_position, np.ndarray)
+                            else monitor_position)
+
+        monitor_rotation = meta.monitor_rotation[:]
+        monitor_rotation = (monitor_rotation.tolist()
+                            if isinstance(monitor_rotation, np.ndarray)
+                            else monitor_rotation)
+
+        camera_position = meta.camera_position[:]
+        camera_position = (camera_position.tolist()
+                           if isinstance(camera_position, np.ndarray)
+                           else camera_position)
+
+        camera_rotation = meta.camera_rotation[:]
+        camera_rotation = (camera_rotation.tolist()
+                           if isinstance(camera_rotation, np.ndarray)
+                           else camera_rotation)
+
+        led_position = meta.led_position[:]
+        led_position = (led_position.tolist()
+                        if isinstance(led_position, np.ndarray)
+                        else led_position)
+
+        rig_geometry = {
+            f"monitor_position_{meta.monitor_position__unit_of_measurement}":
```

Comment:
Given that `RigGeometry` has hardcoded the units in the argument names, instantiating `RigGeometry` will fail if the units don't match

Reply:
Sure. But I can imagine that error message being a little more confusing than an error emitted from this part of the code that says "Your coordinates are in cm; they need to be in mm", etc.

Reply:
If you look at `to_nwb`, the units are hardcoded; in LIMS, the column names have hardcoded suffix including the units; I don't think there's need for validation here because of that.

Reply:
I'm worried about what happens if our conventions change in the future (or, worse, if someone tries to roll their own NWB file and run this `from_nwb` on that).

Maybe the root of all this is that I'm not a fan of the keys in this dict giving you the impression that you can choose a different set of units when, in fact, you cannot. It also makes this part of the code hard to read, in that you can't technically tell which kwargs are getting passed to the `RigGeometry` constructor without going back to that constructor and looking at what kwargs it expects.

Reply:
I have removed the dynamic keys